src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,41 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MINIMUM_PASSWORD_LENGTH = 6;
+    private static final int MINIMUM_LOGIN_LENGTH = 6;
+    private static final int MINIMUM_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user.getLogin() == null) {
+            throw new InvalidUserDataException("Login can't be null");
+        }
+        if (user.getPassword() == null) {
+            throw new InvalidUserDataException("Password can't be null.");
+        }
+        for (User person : Storage.people) {
+            if (person.getLogin().equals(user.getLogin())) {
+                throw new InvalidUserDataException("You cannot add two users with the"
+                        + " same login to the storage.");
+            }
```

Comment:
Iterating through the storage directly to check for duplicate logins can lead to performance issues if the storage grows large. Consider adding a method in the StorageDao to check for existing logins, which can be optimized separately.